deploy/buildtools/buildfarm.py

```diff
@@ -0,0 +1,356 @@
+        self.spot_max_price = self.args['spot-max-price']
+
+        self.dest_build_dir = self.args["build-dir"]
+        if not self.dest_build_dir:
+            raise Exception("ERROR: Invalid null build dir")
+
+    def request_build_host(self, build_config: BuildConfig) -> None:
+        """Launch an AWS EC2 instance for the build config.
+
+        Args:
+            build_config: Build config to request build host for.
+        """
+
+        # get access to the runfarmprefix, which we will apply to build
+        # instances too now.
+        aws_resource_names_dict = aws_resource_names()
+        # just duplicate the runfarmprefix for now. This can be None,
+        # in which case we give an empty build farm prefix
+        build_farm_prefix = aws_resource_names_dict['runfarmprefix']
+
+        buildfarmprefix = '' if build_farm_prefix is None else build_farm_prefix
+
+        inst_obj = launch_instances(
+            self.instance_type,
+            1,
+            self.build_instance_market,
+            self.spot_interruption_behavior,
+            self.spot_max_price,
+            blockdevices=[
+                {
+                    'DeviceName': '/dev/sda1',
+                    'Ebs': {
+                        'VolumeSize': 200,
+                        'VolumeType': 'gp2',
+                    },
+                },
+            ],
+            tags={ 'fsimbuildcluster': buildfarmprefix },
+            randomsubnet=True)[0]
+
+        self.build_hosts.append(EC2BuildHost(build_config=build_config, inst_obj=inst_obj, dest_build_dir=self.dest_build_dir))
+
+    def wait_on_build_host_initialization(self, build_config: BuildConfig) -> None:
+        """Wait for EC2 instance launch.
+
+        Args:
+            build_config: Build config used to find build host that must ready.
+        """
+        build_host = cast(EC2BuildHost, self.get_build_host(build_config))
+        wait_on_instance_launches([build_host.launched_instance_object])
+        build_host.ip_address = build_host.launched_instance_object.private_ip_address
+
+    def release_build_host(self, build_config: BuildConfig) -> None:
+        """ Terminate the EC2 instance running this build.
+
+        Args:
+            build_config: Build config to find build host to terminate.
+        """
+        build_host = cast(EC2BuildHost, self.get_build_host(build_config))
+        instance_ids = get_instance_ids_for_instances([build_host.launched_instance_object])
+        rootLogger.info(f"Terminating build instance {instance_ids}. Please confirm in your AWS Management Console")
+        terminate_instances(instance_ids, dryrun=False)
+
+    def __repr__(self) -> str:
+        return f"{type(self)}(NAME={self.NAME()}, build_hosts={self.build_hosts!r} instance_type={self.instance_type} build_instance_market={self.build_instance_market} spot_interruption_behavior={self.spot_interruption_behavior} spot_max_price={self.spot_max_price})"
```

Comment:
Since you can't literally pass this into the REPL, the python convention is to wrap the entire repr in `<>`
```suggestion
        return f"< {type(self)}(NAME={self.NAME()}, build_hosts={self.build_hosts!r} instance_type={self.instance_type} build_instance_market={self.build_instance_market} spot_interruption_behavior={self.spot_interruption_behavior} spot_max_price={self.spot_max_price}) >"
```